### PR TITLE
Migrate to redhat ubi9 base image

### PR DESCRIPTION
**Pull request by pcrane**
## What type of PR is this? (check all applicable)

- [x] Other

## Description

Migrates the redash image to use RedHat UBI9 as a base image.

### CODE REVIEW

File: tests/extensions/test_extensions.py [hunk 1]
```diff
@@ -71,6 +72,7 @@ def setUpClass(cls):
     def tearDownClass(cls):
         subprocess.call(["pip", "uninstall", "-y", "redash-dummy"])
 
+    @unittest.skip
```
Comment (jamescatto) [inline tests/extensions/test_extensions.py:75]: Is this to avoid something breaking?

File: requirements_all_ds.txt [hunk 2]
```diff
@@ -8,21 +8,18 @@ pyhive==0.6.1
...
 botocore==1.27.85
 sasl>=0.1.3
 thrift>=0.8.0
 thrift_sasl>=0.1.0
-cassandra-driver==3.21.0
 atsd_client==3.0.5
 simple_salesforce==0.74.3
 PyAthena>=1.5.0
-pymapd==0.26.0
+pyomnisci==0.28.2
```
Comment (jamescatto) [inline requirements_all_ds.txt:21]: does this replace `pymapd` ?

File: .github/workflows/build-and-publish-docker-image.yml [hunk 3]
```diff
@@ -27,6 +27,8 @@ jobs:
           tags: |
             type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'master') }}
             type=raw,value=10.1.1,enable=${{ github.ref == format('refs/heads/release/{0}', '10.0.x') }}
+            type=raw,value=10.1.1-ubi,enable=${{ github.ref == format('refs/heads/release/{0}', '10.0.x') }}
```
Comment (jamescatto) [inline .github/workflows/build-and-publish-docker-image.yml:30]: why are there 2?

**Review by jamescatto — APPROVED** [review]
It seems like there is a lot of packages removed. Presumably they are only required at runtime or are part of the `ubi9` image by default

File: tests/extensions/test_extensions.py (hunk 1 above)
Comment (pcrane) [reply tests/extensions/test_extensions.py:75]: The problem seems to be the file/folder it's looking for doesn't get created. I'm not that fussed as, IIRC, it's some front-end thing.

File: requirements_all_ds.txt (hunk 2 above)
Comment (pcrane) [reply requirements_all_ds.txt:21]: Correct

File: .github/workflows/build-and-publish-docker-image.yml (hunk 3 above)
Comment (pcrane) [reply .github/workflows/build-and-publish-docker-image.yml:30]: Ones just ubi, the other is ubi9.

Comment (jamescatto) [reply .github/workflows/build-and-publish-docker-image.yml:30]: Sure, but there isn't a difference between them or is there? If not why have them? Which one should be used?

Comment (pcrane) [reply .github/workflows/build-and-publish-docker-image.yml:30]: It follows a similar pattern to the other offerings on docker hub. It's a convenience thing - either/or is fine.

**Pull request merged**